### ISSUE TITLE
[feature request] CSV like output

### ISSUE THREAD

**Issue by tamodolo** [anidl/multi-downloader-nx#601]
### Type

CLI

### Suggestion

I'm trying to do an script that handle series I'm watching and check new episodes on one command. I had something that worked really great with yt-dlp but reading the output of this software is a nightmare. Because of that I would humble ask if there is a possibility to dump search results based on alias like it's done in filename scheme.

for example: I'd like to check if new episodes of a season is available. So I pass the anidl somewhat like this:

anidl --service crunchy --series [code] --file-dump $alias1 $alias2 $etc --output-file-name name.csv

a csv file is a lot easier to parse and I could do something that could use this to easely get info around with one click.

A way to do the same that produce an easy to read by machine would help a lot too.

Thanks!

**Comment by Jaynator495** [comment]
True, that would be a nice feature. Though we actually have something like this built in called the archive. Using `--addArchive`, you can save every series you download to the archive, so when you run `--downloadArchive` with `--but` it will download all the new episodes of a show. I suspect a lot of refactoring would have to be done to output as a csv, json might be a little more manageable though

**Comment by tamodolo** [comment]
> True, that would be a nice feature. Though we actually have something like this built in called the archive. Using `--addArchive`, you can save every series you download to the archive, so when you run `--downloadArchive` with `--but` it will download all the new episodes of a show.
> 
> As for the actual output, I suspect a lot of refactoring would have to be done to output as a csv, json might be a little more manageable though

json is still easier and would acomplish the objective completly. There's no need of CSV if a good json is outputed.

about --downloadArchive, I was testing it and didn't understand it entirely just with documentation. When I add a series anidl tries to download everything... so I really don't know if it's possible to add for example just download the dub version with the .ass subs for that dub (not the complete one). So I don't think how I could use it to that without download everything else.

**Comment by Jaynator495** [comment]
So I was thinking of this over dinner, and would the raw JSON data from the API work (This would be the easiest to accomplish)? Or do you want something more custom? Is there a structure you would prefer? In the example you gave (series), I was thinking something like this:
```json
{
  "series": {rawCrunchySeries},
  "seasons": [{rawCrunchySeason}, {rawCrunchySeason}]
}
```
In the seasons would be the episodes from the API. Honestly making a full TypeScript implementation of their API may not be a bad idea - but a bit off topic.

Circling back to the archive though, basically everything you download (so long as you have `--addArchive` in the command or `addArchive: true`) will be added to the archive. It doesn't really handle multiple languages though, you could specify the dub you want to download with --dubLang when using --downloadArchive, but, it would be added to the archive if addArchive is present, preventing a download in another dub at another time. One day I have to completely rewrite how archiving works, but I haven't gotten around to it yet.

**Comment by tamodolo** [comment]
It doesn't need to be custom. If it's easier to just drop the json from the api for you then that's enough for me too. All info will come from there anyway and that can potentialy opens development for stand alone GUIs that can hook info from this program if a dump of the json can be made for any input.

**Comment by stratumadev** [comment]
JSON output added
